module/VuFind/src/VuFind/Hierarchy/TreeDataSource/PluginManager.php

```diff
@@ -38,6 +38,28 @@
  */
 class PluginManager extends \VuFind\ServiceManager\AbstractPluginManager
 {
+    /**
+     * Default plugin aliases.
+     *
+     * @var array
+     */
+    protected $aliases = [
+        'solr' => 'VuFind\Hierarchy\TreeDataSource\Solr',
+        'xmlfile' => 'VuFind\Hierarchy\TreeDataSource\XMLFile',
+    ];
+
+    /**
+     * Default plugin factories.
+     *
+     * @var array
+     */
+    protected $factories = [
+        'VuFind\Hierarchy\TreeDataSource\Solr' =>
+            'VuFind\Hierarchy\TreeDataSource\Factory::getSolr',
```

Comment:
Might be worth replacing this static factory with a non-static one (i.e. SolrFactory, implementing FactoryInterface).